workflow/controller/operator.go

```diff
@@ -501,20 +502,26 @@ func (woc *wfOperationCtx) persistUpdates() {
 	} else {
 		woc.wf = wf
 	}
+	// After we successfully persist an update to the workflow, the informer's
+	// cache is now invalid. It's very common that we will need to immediately re-operate on a
+	// workflow due to queuing by the pod workers, so we update the informer.
```

Comment:
Very very clever.

Reply:
We should spend some time thinking to guarantee that there won't be any race conditions or unsynced informers if we manually rewrite back to them